doc/source/whatsnew/v1.0.0.rst

```diff
@@ -174,6 +174,43 @@ Backwards incompatible API changes
 
    pd.arrays.IntervalArray.from_tuples([(0, 1), (2, 3)])
 
+.. _whatsnew_1000.api_breaking.GroupBy.apply:
+
+``GroupBy.apply`` behaves consistently with `as_index`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- The result of :meth:`GroupBy.apply` sometimes contained the grouper column(s),
```

Comment:
Can you add an example where `GroupBy.apply` is used to essentially concat groups together? I don't think many will find this example controversial, but doing something like `df.groupby("a").apply(lambda x: x.iloc[0])` may surprise some going forward if they previously expected the grouping to be included as part of the apply output